### ISSUE TITLE
Support SVG in rcynic.py graphics

### ISSUE THREAD

**Issue by sraustein** [dragonresearch/rpki.net#266]
Right now the rrdtool output in rcynic.py --show-graphs mode is PNG.  SVG is a better format for this for various reasons, including that it would let the user zoom in on the image to examine details that are hard to see otherwise.

Sadly, the state of SVG support in browsers is still a nightmare.

In theory, all modern browsers support <object type="image/svg+xml"/>.  In practice, Firefox has an unhealthy tendency to gag on this.

Firefox doesn't even try to support <a/> elements containing SVG images; if one wants to link from an SVG image one has to edit the SVG (admittedly, not difficult, it's just XML) to add an <a/> element to the SVG itself -- but even then the link may not do what you want, as the browser may decide to render the target of the link inside the <object/> element, so you get a page nested within another page.

The Net of a Million Lies says that it's easier just to use <img/> with SVG, but that's non-standard and may not work at all.

As far as I can tell, this morass means that the only halfway sane way we could use SVG would be:
- Render all images as both PNG and SVG
- Make all the images on the blah_graph.html sub-pages be links to the SVG.
- Use PNG with <img/> for all the images embedded on the HTML pages.

In other words, SVG would become an additional set of images rather than replacing anything in the current PNG collection.

Might be worth it.  Could even use different image sizes to make the SVG more detailed, think of the PNG as a big thumbnail for the SVG.  Just kind of frustrating that SVG is so crippled, it's 98% the right thing but that last two percent is a showstopper.

At any rate: I'm attaching the work I did trying to use SVG as a drop-in replacement for PNG.  It doesn't work, for the reasons described above, but perhaps someday the browsers will get smart enough that it does work.  Or something.

_Trac ticket #250 component rcynic priority minor, owner sra, created by sra on 2012-05-16T16:17:53Z, last modified 2012-06-27T19:38:44Z_
